core/http-auth/src/main/java/software/amazon/awssdk/http/auth/internal/DefaultAwsV4QueryHttpSigner.java

```diff
@@ -31,11 +55,135 @@ public final class DefaultAwsV4QueryHttpSigner implements AwsV4QueryHttpSigner {
 
     @Override
     public SyncSignedRequest sign(SyncSignRequest<? extends AwsCredentialsIdentity> request) {
-        throw new UnsupportedOperationException();
+        // anonymous credentials, don't sign
+        if (CredentialUtils.isAnonymous(request.identity())) {
+            return SyncSignedRequest.builder()
+                .request(request.request())
+                .payload(request.payload().orElse(null))
+                .build();
+        }
+
+        String contentHash = calculateContentHash(request.payload().orElse(null), null);
+
+        SdkHttpRequest signedRequest = doSign(request, contentHash).build();
+        return SyncSignedRequest.builder()
+            .request(signedRequest)
+            .payload(request.payload().orElse(null))
+            .build();
+    }
+
+    private SdkHttpRequest.Builder doSign(SignRequest<?, ? extends AwsCredentialsIdentity> request, String contentHash) {
```

Comment:
I'm still considering whether this can be done via signer properties instead of different interface/impl with lot of duplication.

Reply:
We can always refactor later if we choose. I don't think the amount of duplication here is harmful, but always open to my mind being changed.

Reply:
With the refactor to make the query signer a sub-type, and the overrides in the query signer being small, we are close to being able to make "query" a signer property. Will discuss more offline.